### PR TITLE
APB-7184 Introduce CBC service

### PR DESCRIPTION
**Pull request by dg-21-hmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/connectors/AgentClientAuthorisationConnector.scala [hunk 1]
```diff
@@ -264,169 +240,42 @@ class AgentClientAuthorisationConnector @Inject()(http: HttpClient)(implicit val
...
-        new URL(
-          baseUrl,
-          s"/agent-client-authorisation/clients/EtmpRegistrationNumber/${pptRef.value}/invitations/received/${invitationId.value}/reject").toString
+    ec: ExecutionContext): Future[Boolean] = {
+    val taxIdType = (service, taxId) match {
+      /* TODO [Service onboarding]
+         another pattern match that could be centralised. But we still have several anomalies which make usage vary...
+         e.g. alt-ITSA (MtdIt with Nino instead of MtdItId)
+         also in some places we use "NI" and elsewhere "NINO" and "ni" :(
+         also sometimes we use "UTR", other times "SAUTR", the upper/lowercase is inconsistent... Sort this out */
```
Comment (CarolineS-QA) [inline app/uk/gov/hmrc/agentinvitationsfrontend/connectors/AgentClientAuthorisationConnector.scala:251]: This is good! SAUTR/UTR I think is a legacy vs not thing? but where possible I agree 😅

File: app/uk/gov/hmrc/agentinvitationsfrontend/connectors/AgentClientAuthorisationConnector.scala [hunk 2]
```diff
@@ -698,6 +566,24 @@ class AgentClientAuthorisationConnector @Inject()(http: HttpClient)(implicit val
...
 
+  def getCbcSubscription(cbcId: CbcId)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[SimpleCbcSubscription]] = {
+    val url = new URL(baseUrl, s"/agent-client-authorisation/cbc/subscriptions/${cbcId.value}").toString
+
+    monitor(s"ConsumedAPI-Get-CBC-subscription-GET") {
+      http.GET[HttpResponse](url).map { response =>
+        response.status match {
+          case OK => Some(response.json.as[SimpleCbcSubscription])
+          case NOT_FOUND =>
+            logger.warn(s"PPT Subscription not found for pptRef: ${cbcId.value}")
```
[the remark this answers is not in the document]
Comment (dg-21-hmrc) [reply app/uk/gov/hmrc/agentinvitationsfrontend/connectors/AgentClientAuthorisationConnector.scala:577]: :eagle: :eyes:

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeauthJourneyController.scala [hunk 3]
```diff
@@ -305,25 +313,14 @@ class AgentLedDeauthJourneyController @Inject()(
...
-      service match {
-        case Service.Vat =>
-          Ok(
-            identifyClientVatView(
-              formWithErrors.or(VatClientForm.form),
-              routes.AgentLedDeauthJourneyController.submitIdentifyVatClient,
-              backLinkFor(breadcrumbs).url,
-              isDeAuthJourney = true
-            ))
-        case Service.Ppt =>
```
Comment (CarolineS-QA) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeauthJourneyController.scala:318]: did you intend to remove this ppt business case? it's not clear it's been replaced/refactored elsewhere

**Review by CarolineS-QA — COMMENTED** [review]
Overall looking good, nearly there 👍

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeauthJourneyController.scala (hunk 3 above)
Comment (dg-21-hmrc) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeauthJourneyController.scala:318]: It is handled later on line 343

**Review by CarolineS-QA — APPROVED** [review]

**Pull request merged**